### PR TITLE
feat(back): improve DX with cors, logs and remove username regex

### LINKED ISSUE #30: Back - Act different in dev

### PR DESCRIPTION
**Pull request by yyewolf**
### What does this PR do?

- Add logging when receiving a request
- Remove username regex
- Add cors when in dev mode

⚠️ You will need to update your `config.yaml` file !

### CODE REVIEW

File: back/api/v1/authentication.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func setupAuthRoutes(api *Api, router *atreugo.Router) {
 }
 
 type LoginRequest struct {
-	Username string `json:"username" validate:"min=2,max=40,regexp=^[a-zA-Z]*$"`
```
Comment (Nounoursdestavernes) [inline back/api/v1/authentication.go:27]: why no regexep ?

**Review by Nounoursdestavernes — COMMENTED** [review]
You know if you are in dev or prod due to the config ?

File: back/api/v1/authentication.go (hunk 1 above)
Comment (yyewolf) [reply back/api/v1/authentication.go:27]: We probably don't want to limit users on their username, just like we won't restrict them on their team names to keep compatibility with CTF Time

Frontend will have to make sure we have no XSS

**Comment by yyewolf** [comment]
> You know if you are in dev or prod due to the config ?

Yep, it has an "environment" that should default to prod

File: back/api/v1/authentication.go (hunk 1 above)
Comment (Nounoursdestavernes) [reply back/api/v1/authentication.go:27]: If possible check in the back and otherwise trust the front

Comment (yyewolf) [reply back/api/v1/authentication.go:27]: It's not "trust the front", rather, the front must not trust the back

Comment (NathanCornelie) [reply back/api/v1/authentication.go:27]: heu , isnt it more like "never trust the front"

**Review by NathanCornelie — APPROVED** [review]
_propre 👍 _

**Pull request merged**